sky/packages/sky/lib/src/widgets/snack_bar.dart

```diff
@@ -71,15 +85,15 @@ class SnackBar extends Component {
     ];
     if (actions != null)
       children.addAll(actions);
-
     return new SlideTransition(
```

Comment:
What if we had a wrapper class that owned the performance and took a transition as a child? So you would do something like:
  return new AnimatedComponent(
    duration: foo,
    direction: bar,
    onDismissed: baz,
    transition: new SlideInTransition()
  );

that way SnackBar doesn't have to be stateful. Obviously not a huge deal here, since SnackBar is the exception regarding statefulness. I'm just thinking it might be a more common pattern.